src/Engine/ProtoCore/DSASM/InstructionSet.cs

```diff
@@ -110,6 +110,112 @@ public struct MetaData
         public int type;
     }
 
+    /// <summary>
+    /// Wrapper over all values handled by the MSIL/CLR engine.
+    /// </summary>
+    internal struct CLRStackValue
+    {
+        // TODO_MSIL: Figure out if we need to use an AddressType, like StackValue
+        public bool IsEnumerable => Type != typeof(string) && ArrayUtils.IsEnumerable(Type);
+
+        public bool IsDouble => Value is double;
+
+        // TODO_MSIL: Figure out how to set/use this flag;
+        public bool IsDefaultArgument => false;
+
+        // TODO_MSIL: Figure out if we need to use an AddressType, like StackValue
+        public bool IsPointer => Type != typeof(string) && !ArrayUtils.IsEnumerable(Type) && !Type.IsValueType && Type.IsClass;
```

Comment:
Not sure I have these covered well enough...
This flag probably needs to be explicitly set during MSILEmit phase (when we get to that...)